packages/flutter/lib/src/widgets/nested_scroll_view.dart

```diff
@@ -284,12 +284,54 @@ class NestedScrollView extends StatefulWidget {
   }
 
   @override
-  _NestedScrollViewState createState() => _NestedScrollViewState();
+  NestedScrollViewState createState() => NestedScrollViewState();
 }
 
-class _NestedScrollViewState extends State<NestedScrollView> {
+/// This is the [State] for [NestedScrollView]. It allows you to obtain the scroll
+/// position of the children of your nested scroll view.
+///
+/// You can access the inner scroll controller via [innerController] and the outer
+/// controller with [outerController], however, notice that the outer scroll position
+/// can also be obtained using [NestedScrollView.controller].
+///
+/// If you want to access the inner scroll controller of your nested scroll view,
+/// you can get its [NestedScrollViewState] by supplying a `GlobalKey<NestedScrollViewState>`
+/// to your [NestedScrollView] widget (the `key` parameter).
+///
+/// {@tool sample}
+///
+/// You would access the [NestedScrollViewState] using a [GlobalKey].
+///
+/// ```dart
+/// @override
+/// Widget build(BuildContext context) {
+///   // You probably want to declare your GlobalKey outside of your build method.
+///   final GlobalKey<NestedScrollViewState> globalKey = GlobalKey();
```

Comment:
This should be outside of the build method. The sample code should show good practises.

Reply:
I tried to do that before, however, the analyzer tests failed doing so. I declared the `globalKey` above the `build` method and the `globalKey` could not be resolved in the build method by the analyzer that goes over the snippets. Is there a sample that shows me how to handle that situation?